packaging_automation/update_docker.py

```diff
@@ -103,48 +110,29 @@ def update_changelog(project_version: str, exec_path: str, postgres_version: str
             raise ValueError(f"Already using version {project_version} in the changelog")
 
 
-def update_all_docker_files(project_version: str, exec_path: str, postgres_version: str):
+def update_all_docker_files(project_version: str, exec_path: str):
     template_path = f"{BASE_PATH}/templates/docker"
     pkgvars_file = f"{exec_path}/pkgvars"
 
-    if postgres_version:
-        update_pkgvars(project_version, template_path, pkgvars_file, postgres_version)
+    postgres_14_version, postgres_13_version, postgres_12_version = read_postgres_versions(pkgvars_file)
 
-    persisted_postgres_version = read_postgres_version(pkgvars_file)
+    latest_postgres_version = postgres_14_version
 
-    update_docker_file_for_latest_postgres(project_version, template_path, exec_path, persisted_postgres_version)
+    update_docker_file_for_latest_postgres(project_version, template_path, exec_path, latest_postgres_version)
     update_regular_docker_compose_file(project_version, template_path, exec_path)
-    update_docker_file_alpine(project_version, template_path, exec_path, persisted_postgres_version)
-    update_docker_file_for_postgres12(project_version, template_path, exec_path)
-    update_docker_file_for_postgres14(project_version, template_path, exec_path)
-    update_changelog(project_version, exec_path, postgres_version)
+    update_docker_file_alpine(project_version, template_path, exec_path, latest_postgres_version)
+    update_docker_file_for_postgres12(project_version, template_path, exec_path, postgres_12_version)
+    update_docker_file_for_postgres13(project_version, template_path, exec_path, postgres_13_version)
+    update_changelog(project_version, exec_path, latest_postgres_version)
 
 
-def read_postgres_version(pkgvars_file: str) -> str:
+def read_postgres_versions(pkgvars_file: str) -> Tuple[str, str, str]:
     if os.path.exists(pkgvars_file):
-        with open(pkgvars_file, "r", encoding=DEFAULT_ENCODING_FOR_FILE_HANDLING,
-                  errors=DEFAULT_UNICODE_ERROR_HANDLER) as reader:
-            lines = reader.readlines()
-            for line in lines:
-                if line.startswith("latest_postgres_version"):
-                    line_parts = line.split("=")
-                    if len(line_parts) != 2:
-                        raise ValueError("keys and values should be seperated with '=' sign")
-                    postgres_version = line_parts[1].rstrip("\n")
-                    break
-            if not postgres_version:
-                raise ValueError("pkgvars file should include a line with key latest_postgres_version")
-    else:
-        # Setting it because pkgvars does not exist initially
-        postgres_version = "13.4"
-    return postgres_version
+        config = dotenv_values(pkgvars_file)
+        return config["postgres_14_version"], config["postgres_13_version"], config["postgres_12_version"]
 
+    return "14.1", "13.5", "12.9"
```

Comment:
can move those into some constant variables in this file ?

```python
DOCKER_DEFAULT_LATEST_VER = "14.1"
DOCKER_DEFAULT_PG13_VER = "13.5"
DOCKER_DEFAULT_PG12_VER = "12.9"
```

Reply:
We could even define them in a brand new python file, then use them both here and also in `test_update_docker.py`.